### PR TITLE
Use get_config() instead of CLI/YAML

### PR DESCRIPTION
**Pull request by vladsaveliev**
Refactoring configuration.

* Use TOML configuration via cpg-utils's `get_config()` in favour of Click and YAML files.
* Removed CLI entirely. Relies on `CPG_CONFIG_PATH` to be set.
* Remove `@pipeline_click_options` decorator, now a pipeline object can be created explicitly with `pipeline = Pipeline()`, as most of the parameters are passed explicitly via global config.
* Removed `RefData`, `Images`, `StorageProvider` objects in favour of the `cpg_utils.hail_batch` functions that use config. 
* Lists of images and references are now located in cpg-utils's `config-template.toml` sections.
 
Related issue: https://github.com/populationgenomics/production-pipelines/issues/59

### CODE REVIEW

File: cpg_pipes/config-template.toml [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+# Datasets to load inputs. If not provided, datasets will be determined 
+# automatically based on the input provider implementation:
+#datasets = []
+
+# Datasets to skip:
+#skip_datasets = []
+
+# Samples to skip:
+skip_samples = [
```
Comment (lgruen) [inline cpg_pipes/config-template.toml:18]: Should this be commented out in the template?

File: cpg_pipes/config-template.toml [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+# Create Seqr ElasticSearch indices for these datasets. If not specified, will
+# create indices for all input datasets.
+#create_es_index_for_datasets = 
+
+[elasticsearch]
+# Configure access to ElasticSearch server
+es_port = 9243
+es_host = 'elasticsearch.es.australia-southeast1.gcp.elastic-cloud.com'
+es_username = 'seqr'
+#es_password =
```
Comment (lgruen) [inline cpg_pipes/config-template.toml:142]:
```suggestion
```

File: cpg_pipes/config-template.toml [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+es_username = 'seqr'
+#es_password = 
+# Load ElasticSearch password from a secret, unless SEQR_ES_PASSWORD is set
+es_password_secret_id = 'seqr-es-password'
+es_password_project_id = 'seqr-308602'
+
+[hail]
+#hail_billing_project = 
+#hail_bucket = 
+hail_pool_label = 'seqr'
```
Comment (lgruen) [inline cpg_pipes/config-template.toml:150]: It's already in the `hail` section:
```suggestion
pool_label = 'seqr'
```

File: cpg_pipes/config-template.toml [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+es_host = 'elasticsearch.es.australia-southeast1.gcp.elastic-cloud.com'
+es_username = 'seqr'
+#es_password = 
+# Load ElasticSearch password from a secret, unless SEQR_ES_PASSWORD is set
+es_password_secret_id = 'seqr-es-password'
+es_password_project_id = 'seqr-308602'
+
+[hail]
+#hail_billing_project = 
+#hail_bucket =
```
Comment (lgruen) [inline cpg_pipes/config-template.toml:149]: Should these have no `hail_` prefix?

File: cpg_pipes/dataproc_scripts/seqr/mt_to_es.py [hunk 5]
```diff
@@ -146,20 +115,25 @@ def _cleanup(es, es_index, es_shards):
...
     """
     Read a GCP secret storing the ES password
     """
-    password = os.environ.get('SEQR_ES_PASSWORD')
-    if password:
+    if password := os.environ.get('SEQR_ES_PASSWORD'):
+        return password
+
+    if password := get_config()['elasticsearch'].get('password'):
         return password
```
Comment (lgruen) [inline cpg_pipes/dataproc_scripts/seqr/mt_to_es.py:126]: These both end up in logs, so we shouldn't store passwords that way.
```suggestion
```

File: cpg_pipes/dataproc_scripts/seqr/mt_to_es.py [hunk 6]
```diff
@@ -146,20 +115,25 @@ def _cleanup(es, es_index, es_shards):
...
+    
+    # Read from a secret.
+    secret_id = get_config()['elasticsearch']['password_secret_id']
+    project_id = get_config()['elasticsearch']['password_project_id']
+
+    client = secretmanager.SecretManagerServiceClient()
+    secret_path = client.secret_version_path(project_id, secret_id, 'latest')
+    # noinspection PyTypeChecker
+    response = client.access_secret_version(request={'name': secret_path})
+    return response.payload.data.decode('UTF-8')
```
Comment (lgruen) [inline cpg_pipes/dataproc_scripts/seqr/mt_to_es.py:136]: You can use `cpg_utils.cloud.read_secret()` instead.

File: cpg_pipes/hb/batch.py [hunk 7]
```diff
@@ -154,31 +158,24 @@ def _print_stat(_d: dict, default_label: str | None = None):
...
     @param description: descriptive name of the Batch (will be displayed in the GUI)
-    @param billing_project: Hail billing project name
-    @param hail_bucket: bucket for Hail Batch intermediate files.
-    @param pool_label: submit jobs to the private pool with this label
     """
-    billing_project = get_billing_project(billing_project)
-    hail_bucket = get_hail_bucket(hail_bucket)
+    billing_project = get_config()['hail']['billing_project']
+    remote_tmpdir = get_config()['hail']['bucket']
+    pool_label = get_config()['hail'].get('hail_pool_label')
```
Comment (lgruen) [inline cpg_pipes/hb/batch.py:169]:
```suggestion
    pool_label = get_config()['hail'].get('pool_label')
```

**Review by lgruen — CHANGES_REQUESTED** [review]

File: cpg_pipes/config-template.toml (hunk 2 above)
Comment (vladsaveliev) [reply cpg_pipes/config-template.toml:142]: I wanted to show that you can provide it explicitly

File: cpg_pipes/config-template.toml (hunk 1 above)
Comment (vladsaveliev) [reply cpg_pipes/config-template.toml:18]: Thanks, fixed!

**Review by lgruen — APPROVED** [review]

**Pull request merged**